### DISCUSSION TITLE
How to identify STEP assembly part containing a 3D point?

### DISCUSSION THREAD

**Discussion by RadekPecher** [Open-Cascade-SAS/OCCT#228] [Q&A]
Dear OCCT community, I am new to this impressive technology and would appreciate your initial advice. Let's say I have a STEP file of an assembly with a couple of parts, labelled by names and different colours. After I read in the file using `STEPCAFControl_Reader`, I would like to identify each part/solid of the assembly for any given point in the 3D space. What would be your recommended (and most efficient) way to do that, please?

**Comment by dpasukhi** [comment]
Hello. Would you like to use visualization part or only using the algorithmic? 
In case if you want to use AIS functionality, you can extend you solution to use XCAFPrs_AISObject and AIS_InteractiveContext::select by mouse position, then Extract the selected object and cast to the object with TDF_Label connection. If you want more details in that way, let me know. But I will need more details about your viewer and how you visualize.
In case if you want pure algo realization, without visualization, then the task is less trivial and require BVH tree for your assembly (that is not part of already prepared algo). If you want more details in that way, let me know.

**Reply by RadekPecher** [reply]
Thanks Dmitry. Yes, I should have added that I'm not interested in the visualisation, but only in the algorithmic identification of the assembly part containing each given point. My objective is to initialise 3D grid cells based on their location in the solids bounded by surfaces from a STEP file. (Will I have to compute and compare line intersections with all those surfaces, or is there any better way to do this or a method ready to be used?)
